controllers/awscluster_controller.go

```diff
@@ -104,26 +104,38 @@ func (r *AWSClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reter
 func reconcileDelete(clusterScope *scope.ClusterScope) (reconcile.Result, error) {
 	clusterScope.Info("Reconciling AWSCluster delete")
 
+	if err := deleteClusterResources(clusterScope); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Cluster is deleted so remove the finalizer.
+	clusterScope.AWSCluster.Finalizers = util.Filter(clusterScope.AWSCluster.Finalizers, infrav1.ClusterFinalizer)
+
+	return reconcile.Result{}, nil
+}
+
+func deleteClusterResources(clusterScope *scope.ClusterScope) error {
+	if clusterScope.VPC().ID == "" {
```

Comment:
This should probably attempt to lookup the VPC (using name and labels) prior to short-circuiting here, otherwise we could potentially orphan a created VPC where persisting the ID back to the AWSCluster resource failed for some reason.

Reply:
@detiber thanks for the review and pointing that out!  I re-did the entire fix.